CMakeLists.txt

```diff
@@ -86,6 +86,9 @@ else()
   message(WARNING "PIE disabled: cmake 3.14+ is required for proper PIE linking support")
 endif()
 
+# TODO sean remove this
+set(CMAKE_CXX_LINK_EXECUTABLE "${CMAKE_CXX_LINK_EXECUTABLE} -ldl")
```

Comment:
This was something that came up when trying to run cmake with `-DCMAKE_BUILD_TYPE=Debug` this made it work but unsure exactly what is going on behind the scenes and whether it should stay here or not